shell/build-jsonnet.sh

```diff
@@ -11,6 +11,19 @@ source "$SCRIPTS_DIR/lib/docker.sh"
 # shellcheck source=./lib/logging.sh
 source "$SCRIPTS_DIR/lib/logging.sh"
 
+# Cache a local copy of the `jsonnet-libs` directory on disk if it doesn't yet exist. We use the SHA
+# of the latest HEAD for invalidation; as HEAD changes with new commits, so will our cache. Do this
+# because it helps us avoid accessing jsonnet-libs via raw.githubusercontent.com, which has
+# aggressive rate limits that we can easily hit. Estimated API usage reduction is +10x since before
+# we'd make 1 request per file (15+ *sonnet files), now we clone at most once per run.
+JSONNET_LIBS_SHA="$(git ls-remote 'https://github.com/getoutreach/jsonnet-libs.git' HEAD | cut -f1)"
+JSONNET_LIBS_CACHE="$HOME/.outreach/.cache/jsonnet-libs/${JSONNET_LIBS_SHA}"
```

Comment:
Would this support other directories too? I could start using this approach for Redis, for example

Reply:
As far as I know, anything in `devbase` is free to stuff whatever they like into `$HOME/.outreach/.cache/` as long as there's no stepping on toes. If you're doing stuff _outside_ devbase, I have no idea if that's _allowed_ or not per-se, but I don't think anyone would mind if you cached stuff in `$HOME/.outreach/.cache/` from whatever tooling codebase you have in mind.

Reply:
@AndrewWinterman Yeah, just don't collide with anything else and you can use `~/.outreach/.cache`.